### PR TITLE
fixed SimCapiValue parsing an empty array as [""]

### PR DESCRIPTION
**Pull request by kiahn-johnston**

### CODE REVIEW

**Comment by kiahn-johnston** [comment]
oh, not everyone has to look, just tagging for visibility!

File: app/scripts/api/snapshot/SimCapiValue.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ define(['check', './SimCapiTypes'], function(check, SimCapiTypes) {
         if (check(value).passive().isBoolean()) {
             return value;
         } else if (check(value).passive().isString()) {
-            return value === 'true' ? true : false;
+            return value === 'true';
```
Comment (lerice) [inline app/scripts/api/snapshot/SimCapiValue.js]: :frog:

**Comment by kiahn-johnston** [comment]
for context, this came up because I am preloading css and hence calling `expose` twice, on the second call it replaces my array with `[""]`

**Comment by lerice** [comment]
ah, just cause i can, and since simcapi needs to go through review to hit master, can you please remove this line:
https://github.com/SmartSparrow/simcapi-js/blob/master/app/scripts/api/snapshot/Transporter.js#L795

it does nothing meaningful anymore and all the tests get console flooded because mocha does a bunch of postMessage's

THANKS :tada:

**Comment by kiahn-johnston** [comment]
Haha, sure!

File: app/scripts/api/snapshot/SimCapiValue.js [hunk 2]
```diff
@@ -36,9 +36,13 @@ define(['check', './SimCapiTypes'], function(check, SimCapiTypes) {
             var newArray = [];
 
             var elements = value.substring(1, value.length - 1).split(',');
+            
+            var isEmpty = elements.length === 1 && elements[0].match(/^\s*$/);
 
-            for (var i = 0; i < elements.length; ++i) {
-                newArray.push(elements[i].trim());
+            if(!isEmpty) {
```
Comment (eliasem) [inline app/scripts/api/snapshot/SimCapiValue.js]: To get rid of the pyramid.

```
if(isEmpty) return newArray;
```

**Comment by eliasem** [comment]
🐘

**Comment by eliasem** [comment]
🐘

File: app/scripts/api/snapshot/SimCapiValue.js (hunk 2 above)
Comment (kiahn-johnston) [inline app/scripts/api/snapshot/SimCapiValue.js]: sure thing.  Do you like `return isEmpty ? [] : elements.map(function(e) { return e.trim(); })` ? or `return isEmpty ? [] : newArray` ?

Comment (lerice) [inline app/scripts/api/snapshot/SimCapiValue.js]: personally i really hate long tricky one liners

Comment (eliasem) [inline app/scripts/api/snapshot/SimCapiValue.js]: I like the `return isEmpty ? [] : newArray;` one. Nice and clean.

Comment (kiahn-johnston) [inline app/scripts/api/snapshot/SimCapiValue.js]: sounds good!

**Comment by kiahn-johnston** [comment]
whoops there is a 🐔  there

**Pull request merged**